File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> set;

        for(int num : nums){
            set[num]++;
        }

        priority_queue<pair<int,int>,vector<pair<int,int>>,greater<pair<int,int>>> minHeap;

        for(auto& num : set){
            minHeap.push({num.second,num.first});
            if(minHeap.size() > k){
                minHeap.pop();
            }
        }

        vector<int> kmost;
        while(!minHeap.empty()){
          kmost.push_back(minHeap.top().second);
          minHeap.pop();
        }

        return kmost;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        int[] output = new int[nums.length];

        if(nums.length == 1){
            return new int[]{nums[0]};
        }

        for(int i=0;i<nums.length;i++){
            int product = 1;
            for(int j=0;j<nums.length;j++){
                if(i != j){
                    product = product*nums[j];
                }
            }
            output[i] = product;
        }
         return output;
    }
}  
